axelar-chains-config/src/utils/sourcify.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+const axios = require('axios');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+async function findValue(obj, contractName, result = []) {
+    if (obj && typeof obj === 'object') {
+        for (const key in obj) {
+            if (key.includes(`/${contractName}.sol`) || key === contractName) {
+                result.push(obj[key].metadata ? obj[key].metadata : obj[key].content);
+            }
+
+            await findValue(obj[key], contractName, result);
+        }
+    }
+}
+
+async function readAndParseFile(filePath, targetKey) {
+    try {
+        const jsonData = await fs.promises.readFile(filePath, { encoding: 'utf8' });
+        const parsedData = JSON.parse(jsonData);
+        const result = [];
+        await findValue(parsedData, targetKey, result);
+        return result;
+    } catch (error) {
+        console.error(`Error reading or parsing file ${filePath}: ${error.message}`);
+        return [];
+    }
+}
+
+async function processDirectory(directoryPath, targetKey) {
+    try {
+        const fileNames = await fs.promises.readdir(directoryPath);
+        const filePaths = fileNames.map((fileName) => path.join(directoryPath, fileName));
+
+        const results = await Promise.all(filePaths.map((filePath) => readAndParseFile(filePath, targetKey)));
+
+        const uniqueResults = [...new Set(results.flat().filter(Boolean))];
+        return uniqueResults;
+    } catch (error) {
+        console.error(`Error reading directory ${directoryPath}: ${error.message}`);
+        return [];
+    }
+}
+
+function searchContractName(rootFolder, targetValue) {
```

Comment:
could also modify getContractPath to accept bytecode instead of contractName and check if it starts with 0x